### PR TITLE
Ensure PlatformView engine life cycle callbacks are invoked

### PR DESCRIPTION
**Pull request by johnmccutchan**
- Move some code off of the message handler onto the parent class.
- Call the engine life cycle callbacks on PlatformView regardless of which mode is used.
- Re-enable and fix test that these callbacks are invoked.

Fixes [#120329](https://github.com/flutter/flutter/issues/120329)

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by reidbaker — COMMENTED** [review]
Most of this pr appears to be moved/reformatted lines of code. Is the meat of the fix [createPlatformView](shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java)'s second to last line where you call maybeInvokeOnFlutterViewAttached?

**Review by reidbaker — APPROVED** [review]

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java [hunk 1]
```diff
@@ -456,204 +456,200 @@ public void clearFocus(int viewId) {
...
-          }
+  // Configures the view for Virtual Display mode, returning the associated texture ID.
+  private long configureForVirtualDisplay(
+      @NonNull PlatformView platformView,
+      @NonNull PlatformViewsChannel.PlatformViewCreationRequest request) {
+    // This mode adds the view to a virtual display, which is wired up to a GL texture that
+    // is composed by the Flutter engine.
+
+    // API level 20 is required to use VirtualDisplay#setSurface.
+    ensureValidAndroidVersion(20);
```
[the remark this answers is not in the document]
Comment (johnmccutchan) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:538]: Agreed. I've renamed it to `enforceMinimumAndroidApiVersion`.

File: shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java [hunk 2]
```diff
@@ -219,7 +219,6 @@ public PresentationState detachState() {
   }
 
   public PlatformView getView() {
```
[the remark this answers is not in the document]
Comment (johnmccutchan) [reply shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java:221]: Done.

File: shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java [hunk 3]
```diff
@@ -456,204 +456,200 @@ public void clearFocus(int viewId) {
...
+  }
 
-        // Configures the view for Hybrid Composition mode.
-        private void configureForHybridComposition(
-            @NonNull PlatformView platformView,
-            @NonNull PlatformViewsChannel.PlatformViewCreationRequest request) {
-          Log.i(TAG, "Using hybrid composition for platform view: " + request.viewId);
-        }
+  // Configures the view for Hybrid Composition mode.
+  private void configureForHybridComposition(
```
[the remark this answers is not in the document]
Comment (johnmccutchan) [reply shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java:524]: I've added a call to `enforceMinimumAndroidApiVersion` at the top of the function. Serving as both documentation and a check.

**Review by reidbaker — APPROVED** [review]

**Pull request merged**